### PR TITLE
build app framework

### PR DESCRIPTION
**Pull request by yaoyimu**
build app framework, including:

1. add home, edit and me tab;
2. access FLEX component for debug;
3. create FCUIKit, and add some convenient methods to the component.

### CODE REVIEW

File: FancyCut/Classes/FCUIKit/Macros/FCScreenMacro.h [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+/// 刘海屏判断
+#define iPhoneX (ABS(MAX(CGRectGetWidth([UIScreen mainScreen].bounds), CGRectGetHeight([UIScreen mainScreen].bounds)) / MIN(CGRectGetWidth([UIScreen mainScreen].bounds), CGRectGetHeight([UIScreen mainScreen].bounds)) - 896 / 414.0) < 0.01 || ABS(MAX(CGRectGetWidth([UIScreen mainScreen].bounds), CGRectGetHeight([UIScreen mainScreen].bounds)) / MIN(CGRectGetWidth([UIScreen mainScreen].bounds), CGRectGetHeight([UIScreen mainScreen].bounds)) - 812 / 375.0) < 0.01)
+
+/// 屏幕宽度
+#define FCScreenWidth ([UIScreen mainScreen].bounds.size.width)
+/// 屏幕高度
+#define FCScreenHeight ([UIScreen mainScreen].bounds.size.height)
+
+#define FCStatusBarHeight (iPhoneX ? 44.0f : 20.0f)
```
Comment (VeggieOrz) [inline FancyCut/Classes/FCUIKit/Macros/FCScreenMacro.h]: This macro is no longer applicable in iPhone 12 and above , it is recommended to delete.

File: FancyCut/Classes/FCUIKit/Macros/FCScreenMacro.h [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+/// 判断是否为 iPad 设备
+#define FCisPadDevice ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad)
+
+/// 判断是否处于竖屏状态
+#define FCIsPortrait                                                                               \
+([[UIApplication sharedApplication] statusBarOrientation] == UIInterfaceOrientationPortrait ||       \
+ [[UIApplication sharedApplication] statusBarOrientation] == UIInterfaceOrientationPortraitUpsideDown)
+
+/// iPad上所有的size乘以1.5
+#define FCiPadScale (1.5)
```
Comment (VeggieOrz) [inline FancyCut/Classes/FCUIKit/Macros/FCScreenMacro.h]: We have no plan for iPad now.

**Review by VeggieOrz — CHANGES_REQUESTED** [review]

File: FancyCut/Classes/FCUIKit/Macros/FCScreenMacro.h (hunk 1 above)
Comment (yaoyimu) [reply FancyCut/Classes/FCUIKit/Macros/FCScreenMacro.h]: already deleted

File: FancyCut/Classes/FCUIKit/Macros/FCScreenMacro.h (hunk 2 above)
Comment (yaoyimu) [reply FancyCut/Classes/FCUIKit/Macros/FCScreenMacro.h]: already deleted!

**Review by VeggieOrz — APPROVED** [review]

**Review by yaoyimu — COMMENTED** [review]
approved

**Pull request merged**